torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1269,3 +1290,36 @@ def _optim_state_dict(
         )
 
     return fsdp_osd
+
+
+def _get_fqn_to_fsdp_param_info(
+    model: nn.Module, dedup_shared_fqns: Set[str]
```

Comment:
The name `dedup_shared_fqns` is a bit unclear to me. Is the deduplication just happening because this is a `set`, but this data structure mainly just represents the FQNs to add as keys to the returned `fqn_to_param_info`?

Reply:
Add a comment. But in general, we only need to track the first fqn of a shared parameter. Same case for `_get_param_to_fqns()` and `param_to_fqns`.

Reply:
After checking the code, I realized this protection is redundant. Please check the comment I added into the code.